src/main/java/app/model/agents/WallFollow/WallFollowAgent.java

```diff
@@ -0,0 +1,417 @@
+            movedForwardLast = true;
+            lastTurn = TurnType.NO_TURN;
+        }
+        else if (noWallDetected(getAngleOfLeftRay()))
+        {
+            if (DEBUG) { System.out.println("ALGORITHM CASE 2"); }
+            newDirection = rotateAgentLeft(true);
+            lastTurn = TurnType.LEFT;
+            movedForwardLast = false;
+        }
+        else if (noWallDetected(direction.getAngle()))
+        {
+            if (DEBUG) { System.out.println("ALGORITHM CASE 3"); }
+            newMove = new Vector(moveLength * direction.getX(), moveLength * direction.getY());
+            movedForwardLast = true;
+            lastTurn = TurnType.NO_TURN;
+        }
+        else
+        {
+            if (DEBUG) { System.out.println("ALGORITHM CASE 4"); }
+            newDirection = rotateAgentLeft(false);
+            lastTurn = TurnType.RIGHT;
+            movedForwardLast = false;
+        }
+        direction = newDirection;  // need to update direction in GameEngine instead?
+        Move move = new Move(newDirection, newMove);
+        updateGraph(move);
+        if (DEBUG)
+        {
+            System.out.println("Did move: " + move.getDeltaPos().getX() + " " + move.getDeltaPos().getY());
+            System.out.println(cellGraph.getVertices());
+        }
+        return move;
+    }
+
+    public void updateNeighbouringVertices(BooleanCell agentCell)
+    {
+        tempAgentCell = agentCell;
+        BooleanCell forwardCell = getNeighbourVertex((int) (agentCell.getX() + direction.getX()*moveLength),
+                (int) (agentCell.getY() + direction.getY()*moveLength), direction.getAngle(), false, 1);
+        BooleanCell leftCell = getNeighbourVertex((int) (agentCell.getX() + rotateAgentLeft(true).getX()*moveLength),
+                (int) (agentCell.getY() + rotateAgentLeft(true).getY()*moveLength),
+                rotateAgentLeft(true).getAngle(), false, 1);
+        BooleanCell rightCell = getNeighbourVertex((int) (agentCell.getX() + rotateAgentLeft(false).getX()*moveLength),
+                (int) (agentCell.getY() + rotateAgentLeft(false).getY()*moveLength),
+                rotateAgentLeft(false).getAngle(), false, 1);
+        cellGraph.updateAgentPos(agentCell,forwardCell,leftCell,rightCell);
+        if (!noMovesDone)
+        {
+            exploreVerticesUntilObstacle(forwardCell, direction, 1);
+            exploreVerticesUntilObstacle(leftCell, rotateAgentLeft(true), 1);
+            exploreVerticesUntilObstacle(rightCell, rotateAgentLeft(false), 1);
+        }
+    }
+
+    public void exploreVerticesUntilObstacle(BooleanCell initVertex, Vector dir, int distFromAgentVertex)
+    {
+        // explore all vertices in a direction until an obstacle is encountered
+        // explore here means agents sees the vertices, not actually moves to them
+        // System.out.println("Checking vertices starting from: " + initVertex.getX() + " " + initVertex.getY());
```

Comment:
Consider leaving this as a Java Doc before the class declaration
```java
/**
 * Java Doc Comments ...
 */
 public void methodName()
 {
```

Also, you have a commented out print ln that should be removed